### DISCUSSION TITLE
Deployments on Vercel failed after vercel cli 28.5.x

### DISCUSSION THREAD

**Discussion by benmkramer** [vercel/vercel#8908] [Help] [answered]
Our deployments have been great until earlier today they started failing for quite strange reasons.

After digging around the errors it was reporting, I noticed quite early in the logs that a big difference between the last successful build and all newer failing builds is the "Vercel CLI" version reported changed from 28.4.17 to 28.5.1. I have tried resetting the code back to that of the most recent successful build and opening a PR to trigger a preview, it also fails in this new way.

Another thing I noticed was that the output directory setting in the github web UI can no longer be saved. We're in a turborepo monorepo and have that output directory set to something like `apps/web/.next` and it has worked great.

We are wondering if https://github.com/vercel/vercel/pull/8742 might be related.

**Comment by benmkramer** [comment]
If there is someway to pin vercel to use CLI 28.4.17 in the meantime that would be great... I have not yet come across any setting/configuration to allow that to happen though.

**Comment by benmkramer** [comment]
Vervet support saved me here in sharing that you can pin to a prior cli version with an environment variable, e.g.

```
VERCEL_CLI_VERSION=vercel@28.4.17
```

It was suggested to not keep this override long term which makes sense, so will leave this discussion open and share more updates towards a resolution as soon as I know more.

**Comment by benmkramer** [comment]
The CLI version used in deploys was rolled back (not sure if that's system wide or if they do that per customer/cohort. Either way, above should still work if having issues with `28.5.x` deploys, and it sounds like they are working on a fix 🙌

**Reply by amyegan** [reply]
Thank you for coming back to share your workaround using `VERCEL_CLI_VERSION` and this update! Confirmed you should no longer need to pin the previous version. The the platform has rolled back to Vercel CLI 28.4.17 for now.

**Comment by amyegan** [comment, marked as answer]
This issue has been resolved. If you still have `VERCEL_CLI_VERSION` set as a workaround, please remember to remove it now so you don't get stuck on an old CLI version 🙇

**Comment by benmkramer** [comment]
Thank you for the follow-up, Amy. The variable was removed before, and
recent builds are running successfully and reporting "Vercel CLI 28.5.5"!

Cheers,
Ben
